slm_lab/agent/algorithm/dqn.py

```diff
@@ -270,7 +276,8 @@ def compute_q_target_values(self, batch):
             end_idx = start_idx + body.action_dim
             _val, q_next_act_b = torch.max(
                 q_next_st_acts[:, start_idx:end_idx], dim=1)
-            logger.debug(f'Q next action for body {body.aeb}: {q_next_act_b.size()}')
+            logger.debug(
+                f'Q next action for body {body.aeb}: {q_next_act_b.size()}')
             q_next_acts.append(q_next_act_b)
             start_idx = end_idx
         # TODO uhh what's this?
```

Comment:
@lgraesser the question from previous PR

Reply:
and another one: https://github.com/kengz/SLM-Lab/pull/33#discussion_r158593883

Reply:
283: Best explained through an example. Suppose we have 2 environments.
1: Action space = 4d, 2: Action space = 2d
When the max is taken in 271, it is taken over a slice from start to end of the action space for each environment. For action space 2 max index of 0 or 1 needs to be shifted to 4 or 5. Without it q_next_st_max_b (286) will have the wrong index. Will open a branch to fix.

Reply:
#33 : This is ok as the last layer of each net is always linear so can be multimodal no problem. Generating a probability distribution to sample from is the responsibility of act_with... fns. These split the eval into each environment to avoid skewing to one tail. See 39 or 68 in algorithm_util.py